### PR TITLE
Added status code on custom exception

### PR DESCRIPTION
**Pull request by estebanl5**

### CODE REVIEW

File: src/Intrinio.Net/IntrinioNetException.cs [hunk 1]
```diff
@@ -20,10 +23,16 @@ public IntrinioNetException(string message, Exception inner)
         {
         }
 
+        public IntrinioNetException(HttpStatusCode statusCode, string message)
+            : base(message)
+        {
+            StatusCode= statusCode;
```
Comment (nathanfranz) [inline src/Intrinio.Net/IntrinioNetException.cs:29]: needs space for equal sign.

File: src/Intrinio.Net.Tests/Api/DataPointTests.cs [hunk 2]
```diff
@@ -20,10 +20,16 @@ public async Task GetDataPointNumberSucceedAsync()
...
 
         [TestMethod]
-        public void GetDataPointNumberBadTickerAsync()
+        public async Task GetDataPointNumberBadTickerAsync()
         {
-            Assert.ThrowsAsync<IntrinioNetException>(
-                async () => await IntrinioTestClient.GetDataPointNumberAsync(BAD_TICKER, MARKET_CAP_TAG));
+            try
+            {
+                _ = await IntrinioTestClient.GetDataPointNumberAsync(BAD_TICKER, MARKET_CAP_TAG);
```
Comment (nathanfranz) [inline src/Intrinio.Net.Tests/Api/DataPointTests.cs:27]: This is fine. Did you check if there was anything cool to chain together with the assert to check the status code tho?

Comment (estebanl5) [reply src/Intrinio.Net.Tests/Api/DataPointTests.cs]: Nice catch let me know what you think about the change.

File: src/Intrinio.Net.Tests/Api/DataPointTests.cs [hunk 3]
```diff
@@ -22,8 +24,10 @@ public async Task GetDataPointNumberSucceedAsync()
         [TestMethod]
         public void GetDataPointNumberBadTickerAsync()
         {
-            Assert.ThrowsAsync<IntrinioNetException>(
-                async () => await IntrinioTestClient.GetDataPointNumberAsync(BAD_TICKER, MARKET_CAP_TAG));
+            IntrinioNetException ex = Assert.ThrowsAsync<IntrinioNetException>(async () => await IntrinioTestClient.GetDataPointNumberAsync(BAD_TICKER, MARKET_CAP_TAG));
```
Comment (nathanfranz) [inline src/Intrinio.Net.Tests/Api/DataPointTests.cs:27]: I like this!

**Review by nathanfranz — APPROVED** [review]

**Pull request merged**